core/src/main/java/hudson/model/View.java

```diff
@@ -1006,6 +1009,36 @@ public synchronized void doDoDelete(StaplerRequest req, StaplerResponse rsp) thr
     public abstract Item doCreateItem( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException;
 
     /**
+     * Makes sure that the given name is good as a job name.
+     * For use from {@code newJob}.
+     */
+    @Restricted(DoNotUse.class) // called from newJob view
+    public FormValidation doCheckJobName(@QueryParameter String value) {
+        // this method can be used to check if a file exists anywhere in the file system,
+        // so it should be protected.
+        getOwner().checkPermission(Item.CREATE);
+
+        if (Util.fixEmpty(value) == null) {
+            return FormValidation.ok();
+        }
+
+        try {
+            Jenkins.checkGoodName(value);
```

Comment:
Well certainly you can do trivial client-side validation like that the field is not empty. But the detailed logic (illegal character names, `Item.CREATE` permission, `ProjectNamingStrategy`, existing child) would be hard to replicate on the client side, and we would wind up with a lot of duplicated code since the server already checks all this stuff before actually creating the new item.